### ISSUE TITLE
weird behavior when the function takes longer than the cron interval

### ISSUE THREAD

**Issue by mzy2240** [OxygenFramework/Oxygen.jl#217]
```julia
module JuliaWebServer 
using Oxygen
using HTTP
using Dates

@get "/greet" function(req::HTTP.Request)
    return "hello world!"
end

@cron "*/2" function error_msg()::Nothing
    sleep(5)
    @info now() "Slept 5 seconds"
    return
end

serve()
end
```

Before running the above example code, I assumed that Oxygen would either wait until the last cron task finish or terminate the last task and kick off a new one, but the actual behavior is somewhat unexpected:

![image](https://github.com/user-attachments/assets/58e1fda9-4351-4aab-814d-f44d93387bf0)

Please note I am only using 1 single thread and I assume the server is running in serial mode. Why the function seems to be running in a background thread? and how to only kick off a new task only if the last one is done?

**Comment by ndortega** [comment]
Hi @mzy2240, 

Sorry for the confusion, let me try to clarify how this works: 

A cron "job" is really just a regular `Task` that's spawned on any available thread. We use the cron expression and the current time to calculate how long we need to sleep for untill it's time to execute. Once it's time to run, we run the given function within its own @async task (which is pined to the thread it's spawned on). This behavior guarantee's the function handler is invoked at the time the cron expession defines regardless of how long each handler takes.  

Because each invocation happens asynchronously, your application is spinning up a new task every 2 seconds and those individual tasks will sleep 5 seconds and then log to the console. If I'm right, you're probably not seeing anything in the console for several seconds until the logs start to pile in after the initial 5 second wait is done. 

Here's the snippet for context:
https://github.com/OxygenFramework/Oxygen.jl/blob/master/src/cron.jl#L102-L107

TLDR; An async task is getting created every 2 seconds which waits 5 seconds before logging to the console

**Comment by mzy2240** [comment]
thanks. now it make sense. so async is the key why it is non-blocking.
